apps/src/accounts/AccountInformation/index.tsx

```diff
@@ -0,0 +1,410 @@
+  showGenderInput,
+  usStateDropdownOptions,
+  countryCode,
+  isUSA,
+}) => {
+  const [name, setName] = useState(userDisplayName ?? '');
+  const [username, setUsername] = useState(userUsername ?? '');
+  const [email, setEmail] = useState(userEmail ?? '');
+  const [gender, setGender] = useState(
+    userProperties?.gender_student_input ?? ''
+  );
+  const [age, setAge] = useState(userAge ?? '');
+  const [usState, setUsState] = useState(userProperties?.us_state ?? '');
+  const [password, setPassword] = useState('');
+  const [passwordConfirmation, setPasswordConfirmation] = useState('');
+  const [currentPassword, setCurrentPassword] = useState('');
+  const [showChangeEmailModal, setShowChangeEmailModal] = useState(false);
+  const [showAccountUpdateSuccess, setShowAccountUpdateSuccess] =
+    useState(false);
+  const [showEmailUpdateSuccess, setShowEmailUpdateSuccess] = useState(false);
+  const [errors, setErrors] = useState<Record<string, string[]>>({});
+
+  const displayNameHelperMessage = useMemo(
+    () =>
+      isStudent ? undefined : i18n.accountInformationDisplayNameHintTeacher(),
+    [isStudent]
+  );
+
+  const emailValue = useMemo(
+    () => (isStudent ? `***${i18n.encrypted()}***` : email ?? ''),
+    [isStudent, email]
+  );
+
+  const usStateOptions = useMemo(
+    () =>
+      [[i18n.accountInformationSelectState(), '']]
+        .concat(usStateDropdownOptions)
+        .map(([text, value]) => ({text, value})),
+    [usStateDropdownOptions]
+  );
+
+  const handleSubmitAccountSettingsUpdate = async () => {
+    resetMessages();
+    setErrors({});
+    const userUpdates = {
+      name,
+      username,
+      password,
+      password_confirmation: passwordConfirmation,
+      current_password: currentPassword,
+      age: isStudent ? age : '21+',
+      gender_student_input: isStudent && gender ? gender : undefined,
+      us_state: isStudent && isUSA ? usState : undefined,
+      country_code: isStudent ? countryCode : undefined,
+    };
+    const response = await fetch('/users', {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+        'X-CSRF-Token': await getAuthenticityToken(),
+      },
+      body: JSON.stringify({
+        user: userUpdates,
+      }),
+    });
+
+    if (response.ok) {
+      setShowAccountUpdateSuccess(true);
+    } else {
+      const validationErrors = await response.json();
+      setErrors(validationErrors);
+    }
+  };
+
+  const handleSubmitEmailUpdate = async ({
+    newEmail,
+    currentPassword,
+    emailOptIn,
+  }: {
+    newEmail: string;
+    currentPassword: string;
+    emailOptIn: string;
+  }) => {
+    resetMessages();
+    const hashedEmail = hashEmail(newEmail);
+    const response = await fetch('/users/email', {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+        'X-CSRF-Token': await getAuthenticityToken(),
```

Comment:
This may already exist on the page and could be fetched with `$('meta[name="csrf-token"]').attr('content')` (I think). If not, this looks good to me.

Reply:
yeah the util function grabs it from there if it's available

![Screenshot 2024-10-02 at 10 48 47 AM](https://github.com/user-attachments/assets/eca7b3d3-bf9b-4424-bf9c-7083bfff4c3c)